helm-chart/README.md

```diff
@@ -44,6 +44,35 @@ connection:
   dbName: timescale
 ```
 
+### User-Managed Connection Parameter Secret
+
+**Note:** We primarily provide this feature for convenience. We suggest that you configure the connection options in `values.yaml` instead. If you are using GitOps and do not want to store database passwords unencrypted in your Git repo, you probably want to use tooling like [helm-secrets](https://github.com/jkroepke/helm-secrets) instead.
+
+If you would like to provide connections details via your own secret which is managed outside the lifecycle of this helm chart, you can do so. The secret must have the same internal structure as the connection configuration described above. Additionally, the `connectionSecretName` parameter of this helm chart must be configured with the name of your secret, and your secret must be in the same namespace as the deployment.
+
+An example of what this could look like is as follows:
+```yaml
+# secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-promscale-connection-secret
+stringData:
+  connection:
+    user: postgres
+    password: ""
+    host: db.timescale.svc.cluster.local
+    port: 5432
+    sslMode: require
+    dbName: timescale
+```
```

Comment:
```suggestion
```
In the values.yml of the helm chart you would then have the following:
```

Reply:
done